.github/scripts/validate_scale_config.py

```diff
@@ -98,47 +137,83 @@ def runner_types_are_equivalent(
 
 
 def is_config_consistent_internally(runner_types: Dict[str, Dict[str, str]]) -> bool:
-    """
-    Ensure that for every linux runner type in the config, there is a corresponding runner type with the
-    prefix "amz2023." that contains all the same settings except for the ami
+    f"""
+    Ensure that for every linux runner type in the config:
+
+    1 - they match RunnerTypeScaleConfig https://github.com/pytorch/test-infra/blob/main/terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/runners.ts#L50
+    2 - they have a max_available of at least 50
+    3 - they have a valid {AMAZON_2023_PREFIX} variant, that overrides only the ami
     """
     errors_found = False
 
-    for runner_type in runner_types:
-        # Skip the windows runner types
-        if runner_type.startswith("windows"):
+    for runner_type, runner_config in runner_types.items():
+        try:
+            jsonschema.validate(runner_config, RUNNER_JSCHEMA)
+        except jsonschema.ValidationError as e:
+            print(f"Runner type {runner_type} has invalid configuration: {e.message}")
+            errors_found = True
+            # continue, as the syntax is invalid and we can't trust the rest of the config
+            # so the next part of the code might break
             continue
 
-        if not runner_type.startswith("amz2023."):
-            base_runner_type = runner_type
-            amz2023_runner_type = f"{AMAZON_2023_PREFIX}{runner_type}"
+        # Ensure that the max_available is at least 50
+        # this is a requirement as scale-up always keeps at minimum some spare runners live, and less than 50
+        # will very easily trigger alerts of not enough runners
+        if runner_config["max_available"] < 50:
+            print(
+                f"Runner type {runner_type} has max_available set to {runner_config['max_available']}, "
+                f"which is less than the minimum required value of 50"
+            )
+            errors_found = True
 
-            if amz2023_runner_type not in runner_types:
-                print(
-                    f"Runner type {base_runner_type} does not have a corresponding {amz2023_runner_type} runner type"
-                )
-                errors_found = True
-                continue
+        # Next validations are for linux runners only
+        if runner_config.get("os").lower() == "windows":
```

Comment:
nit: less likely to conflict with future updates, like adding mac runners one day

```suggestion
        if runner_config.get("os").lower() != "linux":
```